packages/flutter/lib/src/cupertino/desktop_text_selection_toolbar_button.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'button.dart';
+import 'colors.dart';
+import 'theme.dart';
+
+// These values were measured from a screenshot of TextEdit on MacOS 10.15.7 on
+// a Macbook Pro.
+const TextStyle _kToolbarButtonFontStyle = TextStyle(
+  inherit: false,
+  fontSize: 14.0,
+  letterSpacing: -0.15,
+  fontWeight: FontWeight.w400,
+  color: CupertinoColors.white,
+);
+
+// This value was measured from a screenshot of TextEdit on MacOS 10.15.7 on a
+// Macbook Pro.
+const EdgeInsets _kToolbarButtonPadding = EdgeInsets.symmetric(
+  vertical: 1.0,
+  horizontal: 20.0,
+);
+
+/// A button in the style of the Mac context menu buttons.
+class CupertinoDesktopTextSelectionToolbarButton extends StatefulWidget {
```

Comment:
Does this really need to be a public class?

Reply:
Not for the purposes of simply creating a Mac toolbar, but I'd argue it's important for allowing users to customize that toolbar easily.  For the Android and iOS text selection toolbars I have also made those buttons public.

I'm happy to hide this if you think that's out of scope for now, though.

Reply:
Update: We've decided to make everything private until we're comfortable with the architecture, then open it up for customization.